test/e2e/config_requests_test.go

```diff
@@ -0,0 +1,84 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/test/framework"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestRequests(t *testing.T) {
+	client := framework.MustNewClientset(t, nil)
+
+	defer framework.MustRemoveImageRegistry(t, client)
+
+	cr := &imageregistryv1.Config{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: imageregistryv1.SchemeGroupVersion.String(),
+			Kind:       "Config",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: imageregistryv1.ImageRegistryResourceName,
+		},
+		Spec: imageregistryv1.ImageRegistrySpec{
+			ManagementState: operatorapi.Managed,
+			Storage: imageregistryv1.ImageRegistryConfigStorage{
+				EmptyDir: &imageregistryv1.ImageRegistryConfigStorageEmptyDir{},
+			},
+			Requests: imageregistryv1.ImageRegistryConfigRequests{
+				Read: imageregistryv1.ImageRegistryConfigRequestsLimits{
+					MaxRunning: 1,
+					MaxInQueue: 2,
+					MaxWaitInQueue: metav1.Duration{
+						Duration: 3 * time.Second,
+					},
+				},
+				Write: imageregistryv1.ImageRegistryConfigRequestsLimits{
+					MaxRunning: 4,
+					MaxInQueue: 5,
+					MaxWaitInQueue: metav1.Duration{
+						Duration: 6 * time.Hour,
+					},
+				},
+			},
+			Replicas: 1,
+		},
+	}
+
+	framework.MustDeployImageRegistry(t, client, cr)
+	framework.MustEnsureImageRegistryIsAvailable(t, client)
+
+	deploy, err := client.Deployments(imageregistryv1.ImageRegistryOperatorNamespace).Get(imageregistryv1.ImageRegistryName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := map[string]string{
+		"REGISTRY_OPENSHIFT_REQUESTS_READ_MAXRUNNING":      "1",
+		"REGISTRY_OPENSHIFT_REQUESTS_READ_MAXINQUEUE":      "2",
+		"REGISTRY_OPENSHIFT_REQUESTS_READ_MAXWAITINQUEUE":  "3s",
+		"REGISTRY_OPENSHIFT_REQUESTS_WRITE_MAXRUNNING":     "4",
+		"REGISTRY_OPENSHIFT_REQUESTS_WRITE_MAXINQUEUE":     "5",
+		"REGISTRY_OPENSHIFT_REQUESTS_WRITE_MAXWAITINQUEUE": "6h0m0s",
+	}
+
+	for _, env := range deploy.Spec.Template.Spec.Containers[0].Env {
```

Comment:
You might be able to use util.CheckEnvVars here (https://github.com/openshift/cluster-image-registry-operator/blob/master/test/framework/util.go), sample usage (https://github.com/openshift/cluster-image-registry-operator/blob/master/test/e2e/gcs_test.go#L120-L128), it will check for existence and value to match.

Reply:
Cool! Thanks for pointing me this, I have updated the PR.